Lombiq.ContentEditors/Assets/Scripts/async-editor/async-editor.js

```diff
@@ -0,0 +1,223 @@
+        }
+    },
+    methods: {
+        initEditor(parameters) {
+            const self = this;
+
+            self.api = new AsyncEditorApiClient(parameters);
+            self.contentId = parameters.contentId;
+            self.editorGroup = parameters.editorGroup;
+            self.defaultErrorText = parameters.defaultErrorText ?? 'Something went wrong.';
+            self.asyncEditorId = parameters.asyncEditorId;
+
+            if (!self.processQuery()) self.loadEditor();
+        },
+        loadEditor(editorGroup) {
+            const self = this;
+
+            self.editorGroup = editorGroup ?? self.editorGroup;
+
+            self.api.loadEditor(
+                self.contentId,
+                self.editorGroup,
+                (success, data) => { self.processApiData(success, data); });
+        },
+        submitEditor(nextEditorGroup) {
+            const self = this;
+            const submittingEditorEvent = new CustomEvent('asyncEditorSubmittingEditor', {
+                bubbles: true,
+                cancelable: true,
+                detail: { asyncEditor: window.asyncEditor }
+            });
+
+            var success = document.dispatchEvent(submittingEditorEvent);
+            if (success) {
+                self.api.submitEditor(
+                    self.contentId,
+                    self.editorGroup,
+                    nextEditorGroup,
+                    new FormData(self.$refs.editorForm),
+                    (success, data) => { self.processApiData(success, data); });
+            }
+        },
+        processApiData(success, data) {
+            const self = this;
+
+            if (success) {
+                const shouldUpdateQuery = self.contentId !== data.contentId || self.editorGroup !== data.editorGroup;
+
+                self.errorText = '';
+                self.validationSummaryHtml = data.validationSummaryHtml;
+                self.contentId = data.contentId;
+                self.editorHtml = data.editorHtml;
+                self.editorGroup = data.editorGroup;
+                self.editorGroups = data.editorGroups;
+                self.scriptsHtml = data.scriptsHtml;
+                self.message = data.message;
+
+                if (shouldUpdateQuery) self.updateQuery();
+            }
+            else {
+                self.errorText = self.defaultErrorText;
+            }
+        },
+        updateQuery() {
+            const self = this;
+
+            const query = { ...self.$route.query };
+            query[self.asyncEditorId + '.contentId'] = self.contentId;
+            query[self.asyncEditorId + '.editorGroup'] = self.editorGroup;
+            router.push({ path: '/', query: query });
+        },
+        processQuery() {
+            const self = this;
+
+            let shouldLoadEditor = false;
+
+            const contentIdKey = self.asyncEditorId + '.contentId';
+            if (!!Object.getOwnPropertyDescriptor(self.$route.query, contentIdKey) &&

```

Comment:
Aren't you just trying to check if the own property exists? If so, use [`hasOwnProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty):
```suggestion
            if (self.$route.query.hasOwnProperty(contentIdKey) &&
```
Same for editorGroupKey below.